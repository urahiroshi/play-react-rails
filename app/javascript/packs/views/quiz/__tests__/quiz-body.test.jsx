import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import QuizBody from '../quiz-body'

const createComponent = () => (
  <QuizBody
    question={{ content: 'hogehoge', answer: 'hoge' }}
    page={2}
    moveNextPage={() => {}}
  />
)

describe('#render', () => {
  it('should show snapshot', () => {
    const wrapper = shallow(createComponent())
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe(
  'when "Check" button clicked after correct answer input',
  () => {
    it('should show "Next Question" button', () => {
      const wrapper = shallow(createComponent())
      wrapper.find('QuestionAnswer').props().onChange('hoge')
      wrapper.find('.quiz-answer-button').props().onClick()
      expect(wrapper.contains('Next Question')).toBe(true)
    })
  },
)

describe(
  'when "Check" button clicked after incorrect answer input',
  () => {
    it('should show "show answer" button', () => {
      const wrapper = shallow(createComponent())
      wrapper.find('QuestionAnswer').props().onChange('fuga')
      wrapper.find('.quiz-answer-button').props().onClick()
      expect(wrapper.find('ShowAnswerButton').exists()).toBe(true)
    })
  },
)
