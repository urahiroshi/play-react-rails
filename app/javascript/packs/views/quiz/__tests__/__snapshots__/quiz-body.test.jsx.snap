// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render should show snapshot 1`] = `
<div
  className="form-horizontal"
>
  <QuestionContent
    content="hogehoge"
    page={2}
  />
  <QuestionAnswer
    answer=""
    onChange={[Function]}
    onEnterKey={[Function]}
  />
  <div
    className="quiz-answer-button-container"
  >
    <AnswerResult />
    <button
      className="btn btn-primary quiz-answer-button"
      onClick={[Function]}
    >
      Check
    </button>
    <div />
  </div>
</div>
`;
